index.py

```diff
@@ -330,28 +350,32 @@ def on_message(client, userdata, message):
     if is_duplicate_event(frigate_event_id):
         return
 
-    snapshot = get_snapshot(frigate_event_id, frigate_url)
-    if not snapshot:
-        return
-
     frigate_plus = config['frigate'].get('frigate_plus', False)
     if frigate_plus and not is_valid_license_plate(after_data):
         return
-
-    license_plate_attribute = get_license_plate(after_data)
-
-    plate_number, plate_score = get_plate(snapshot, after_data, license_plate_attribute)
-    if not plate_number:
+    
+    snapshot = get_snapshot(frigate_event_id, frigate_url, True)
+    if not snapshot:
         return
 
-    start_time = datetime.fromtimestamp(after_data['start_time'])
-    formatted_start_time = start_time.strftime("%Y-%m-%d %H:%M:%S")
-
-    store_plate_in_db(plate_number, plate_score, frigate_event_id, after_data, formatted_start_time)
-    set_sublabel(frigate_url, frigate_event_id, plate_number, plate_score)
+    plate_number, plate_score = get_plate(snapshot, after_data)
+    if plate_number:
+        start_time = datetime.fromtimestamp(after_data['start_time'])
+        formatted_start_time = start_time.strftime("%Y-%m-%d %H:%M:%S")
 
-    send_mqtt_message(plate_number, plate_score, frigate_event_id, after_data, formatted_start_time)
+        store_plate_in_db(plate_number, plate_score, frigate_event_id, after_data, formatted_start_time)
+        set_sublabel(frigate_url, frigate_event_id, plate_number, plate_score)
 
+        send_mqtt_message(plate_number, plate_score, frigate_event_id, after_data, formatted_start_time)
+    
+    if plate_number or config['frigate'].get('always_save_snapshot', False):
```

Comment:
the reason this was up above because of `always_save_snapshot` - if this is false then we wont always save snapshot 

```
frigate_plus = config['frigate'].get('frigate_plus', False)
    if frigate_plus and not is_valid_license_plate(after_data):
        return
```

Reply:
I took the view that if `is_valid_license_plate` fails, it's because Frigate didn't find a valid plate so there's no need to save a snapshot - it's only if there IS a plate but platerecognizer fails to read it that we still want the snapshot to be saved.  In that case, it works here OK.